### ISSUE TITLE
"ValueError: too many values to unpack" when querying indexes with more that one _

### ISSUE THREAD

**Issue by agitator** [zopefoundation/Products.ZCatalog#114]
While updating a Plone 5.0.4 site to 5.2.3 I ran into this Error

```
  Module plone.api.content, line 664, in find
  Module Products.CMFPlone.CatalogTool, line 463, in searchResults
  Module Products.ZCatalog.ZCatalog, line 625, in searchResults
  Module Products.ZCatalog.Catalog, line 1091, in searchResults
  Module Products.ZCatalog.Catalog, line 634, in search
  Module Products.ZCatalog.Catalog, line 562, in _search_index
  Module Products.ZCatalog.query, line 91, in __init__
ValueError: too many values to unpack
....

(Pdb++) request
{'prioritization_lv1_effective': datetime.datetime(2021, 1, 22, 14, 6, 8, 965079), 'allowedRolesAndUsers': ['Anonymous', 'Authenticated', 'Manager', 'user:admin'], 'sort_on': ['effective'], 'sort_order': 'reverse', 'prioritization_lv1': True}
(Pdb++) field
'prioritization_lv1_effective'
(Pdb++) field.split('_')
['prioritization', 'lv1', 'effective']

```

Looks like the changes in https://github.com/zopefoundation/Products.ZCatalog/commit/64b1f3ba7ae1e598e65006d68e6f1d8258ae66b0 cause this error.

**Comment by jensens** [comment]
Probably an `field.lsplit('_', 1)` would fix it?

**Comment by petschki** [comment]
@jensens I think `field.rsplit('_', 1)` is the correct fix. only the last part is allowed as index operator.

**Comment by petschki** [comment]
just found this open PR: https://github.com/zopefoundation/Products.ZCatalog/pull/79 ... if this gets updated with `rsplit` we should all be happy here

/cc @thet

**Comment by icemac** [comment]
Fixed in #116.
